cnxdb/migrations/20170911201035_transform_cnxml_to_html.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+import logging
+import time
+from datetime import timedelta
+
+import rhaptos.cnxmlutils
+from dbmigrator import deferred, logger
+
+
+def _batcher(seq, size):
+    for pos in range(0, len(seq), size):
+         yield seq[pos:pos + size]
+
+
+def should_run(cursor, limit='LIMIT 1'):
+    version = rhaptos.cnxmlutils.__version__
+    version_text = '%data-cnxml-to-html-ver="{}"%'.format(version)
+    logger.info('Looking for {}'.format(version_text))
+    cursor.execute("""\
+SELECT files.fileid, max(module_ident)
+    FROM module_files
+    NATURAL JOIN files
+    WHERE filename = 'index.cnxml.html'
+      AND convert_from(file, 'utf-8') NOT LIKE %s
+    GROUP BY files.fileid {}""".format(limit),
```

Comment:
We may want to build an index on something like `stringpos(convert_from(file, 'utf-8'), version_text)`, then update the index whenever we rollout new rhaptos.cnxmlutils, and test via `stringpos(convert_from(file, 'utf-8'), version_text) = 0` That way as conversions happen, the test gets cheaper and cheaper - the index contains fewer zero entries.

Reply:
while an index makes it a lot faster, this approach takes approx. 5-6 sec. (index supported scan takes ~ 130 ms.) so certainly fast enough.

Reply:
I'll leave it then.